source/security/encryption/encryption-key-management.rst

```diff
@@ -15,12 +17,223 @@ Server-Side Object Encryption
+   <ServerSideEncryptionCustomerKeys.html>`.
+
+.. _minio-encryption-sse-encryption-process:
+
+Encryption Process Overview
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MinIO uses three distinct keys when performing server-side encryption or
+decryption:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 30 70
+
+   * - Key
+     - Description
+
+   * - External Key (EK) 
+     - An external secret key used to generate and encrypt additional encryption
+       keys. 
+      
+       - For SSE-C, the EK is the client-provided encryption key. MinIO
+         never stores this key to disk and requires the application to perform
+         all key management operations.
+
+       - For SSE-S3, the EK is generated by :minio-git:`MinIO KES </kes>` using
+         a Customer Master Key (CMK) stored in the configured external KMS. 
+         KES returns a plain-text representation of the EK for performing
+         cryptographic operations, and a CMK-encrypted representation of the
+         EK for storage alongside the object metadata.
+         
+         See the :minio-git:`KES Wiki </kes/wiki>` for more information on
+         configuring KES.
+
+       MinIO cannot decrypt objects encrypted using SSE-C or SSE-S3 without the
+       associated EK or CMK respectively. Disabling or deleting these keys
+       results in associated encrypted objects being rendered unreadable. See
+       :ref:`minio-encryption-sse-secure-erasure-locking` for more information.
+
+   * - Key Encryption Key (KEK)
+     - A 256-bit unique random secret key deterministically generated by MinIO
+       at the time of each cryptographic operation. MinIO never stores the KEK
+       on disk.
+
+       The key-derivation algorithm uses a pseudo-random function 
+       (:ref:`PRF <minio-encryption-sse-primitives>`) that
+       takes the EK, a randomly generated initialization vector, and
+       a context consisting of values like the bucket and object name.
+
+   * - Object Encryption Key (OEK) 
+     - A 256-bit unique random secret key used to encrypt or decrypt an object.
+       MinIO generates an OEK when first encrypting an object. MinIO encrypts
+       the OEK using the KEK and stores the encrypted OEK as metadata with the
+       object.
+
+       MinIO decrypts the OEK into RAM as part of cryptographic operations. 
+       MinIO never stores the OEK on disk in an unecrypted format.
+
+All secret keys generated by MinIO are 256-bits long. MinIO uses a
+cryptographically secure pseudo-random generator (CSPRNG) as part of key
+generation. In the context of SSE, the MinIO CSPRNG only enforces that the
+generated values are unique rather than truly random.
+
+The following diagrams describe the key hierarchy built by MinIO
+for each object. Each tab corresponds to the specific hierarchy for 
+SSE-S3 and SSE-C respectively:
+
+.. tabs::
+   
+   .. tab:: SSE-S3
+
+      .. image:: /images/minio-encryption-sse-s3-key-hierarchy.svg
```

Comment:
Minor nit, since this graphic is a bit dense it would be good to be able to zoom in without opening a new page, not a showstopper but a nice to have

Reply:
OH good call - I thought I had installed an extension to fix that, but I might not have done so properly. I'll explore adding a lightbox to make this behavior more consistent across the docs.